teacherchannel/module.py

```diff
@@ -0,0 +1,257 @@
+        )
+
+    @tasks.loop(seconds=10.0, count=1)
+    async def load_deltas(self):
+        guild_ids = TeacherChannelDB.get_guild_ids()
+        for guild_id in guild_ids:
+            guild = self.bot.get_guild(guild_id)
+            channels_to_sync = TeacherChannelDB.get_all(guild_id)
+            for teacherchannel in channels_to_sync:
+                master = guild.get_channel(teacherchannel.master_id)
+                slave = guild.get_channel(teacherchannel.slave_id)
+                if not master:
+                    if slave is not None:
+                        name_to_log = slave.name
+                        await slave.delete()
+                    else:
+                        name_to_log = "<#" + str(teacherchannel.slave_id) + ">"
+                    await guild_log.warning(
+                        None,
+                        None,
+                        f"Teacher channel {name_to_log} cannot be synced, channel #<{teacherchannel.master_id}> "
+                        f"does not exist! Deleting from db.",
+                    )
+                    teacherchannel.remove_channel()
+                    continue
+                if not slave:
+                    await guild_log.warning(
+                        None,
+                        None,
+                        f"Teacher channel #<{teacherchannel.slave_id}> (slave of #{master.name}) cannot be synced, "
+                        f"it does not exist! Deleting from db.",
+                    )
+                    teacherchannel.remove_channel()
+                    continue
+                await self._sync(slave, master, teacherchannel)
+
+    @load_deltas.before_loop
+    async def before_load(self):
+        """Ensures that bot is ready before syncing channels."""
+        await self.bot.wait_until_ready()
+
+    @commands.Cog.listener()
+    async def on_guild_channel_update(
+        self,
+        channel_before: nextcord.abc.GuildChannel,
+        channel_after: nextcord.abc.GuildChannel,
+    ):
+        if not isinstance(channel_after, nextcord.TextChannel):
+            return
+        teacherchannel = TeacherChannelDB.get(channel_after.guild.id, channel_after.id)
+        if not teacherchannel:
+            return
+        if channel_after.id == teacherchannel.slave_id:
+            return
+        await self._sync(channel_before, channel_after, teacherchannel)
+
+    async def _sync(
+        self,
+        channel_before: nextcord.abc.GuildChannel,
+        channel_after: nextcord.abc.GuildChannel,
+        teacherchannel: TeacherChannelDB,
+    ) -> None:
+        slave_channel = channel_after.guild.get_channel(teacherchannel.slave_id)
+        if not slave_channel:
+            await guild_log.warning(
+                None,
+                None,
+                f"Teacher channel #<{teacherchannel.slave_id}> cannot be synced, it does not exist! Deleting from db.",
```

Comment:
Again, same as above. And split the string to two lines please, this is too long; try to keep it under 100.